### PR TITLE
ci: Re-enable all linters and tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-416

**Changes:**
- Multiple minor fixes to make all unit tests working again, fixed compatibility with Windows OS.

-----------

### CODE REVIEW

File: internal/pkg/service/common/etcdop/mutex.go [hunk 1]
```diff
@@ -292,9 +294,9 @@ func (m *activeMutex) localUnlock() error {
 func (m *activeMutex) dbLock(ctx context.Context) (err error) {
 	// Get session at the first time
 	if m.dbSession == nil {
-		m.dbSession, err = m.store.session.Session()
+		m.dbSession, err = m.store.session.WaitForSession(ctx)
 		if err != nil {
-			return err
+			return errors.PrefixError(err, "cannot get concurrency session")
 		}
```
Comment (michaljurecko) [inline internal/pkg/service/common/etcdop/mutex.go:300]: We should wait for the session, until the context is cancelled.

It is edge-case, when the connection is broken - I noticed it in CI, because there are fewer resources, and etcd is sometimes overloaded with parallel tests - it doesn't respond for a while.
​

File: internal/pkg/service/common/etcdop/mutex_test.go [hunk 2]
```diff
@@ -162,11 +162,11 @@ func TestMutex_ParallelWork(t *testing.T) {
 				lockTesters = append(lockTesters, lockTester{
 					LockName: fmt.Sprintf("locks/my-lock-%02d", i+1),
 					CriticalWork: func(mtx *Mutex) {
-						assert.NoError(t, mtx.Lock(ctx))
+						require.NoError(t, mtx.Lock(ctx))
```
Comment (michaljurecko) [inline internal/pkg/service/common/etcdop/mutex_test.go:165]: Stop on error.

File: internal/pkg/service/stream/storage/level/local/dir.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+func NormalizeDirPath(path string) string {
+	// Remove special characters, Windows prohibits special characters in directory names
+	path = regexpcache.MustCompile(`\.|:`).ReplaceAllString(path, "-")
+
+	// Fix path separator on Windows
+	path = filepath.FromSlash(path) //nolint:forbidigo
+
+	return path
+}
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/dir.go:17]: Helper to remove Windows sensitive characters from the filesystem path.

File: internal/pkg/utils/etcdhelper/fortest.go [hunk 4]
```diff
@@ -116,7 +116,7 @@ func clientForTest(t testOrBenchmark, ctx context.Context, cfg etcdclient.Config
 	etcdClient, err := etcd.New(etcd.Config{
 		Context:              ctx,
 		Endpoints:            []string{cfg.Endpoint},
-		DialTimeout:          5 * time.Second,
+		DialTimeout:          15 * time.Second,
 		DialKeepAliveTimeout: 5 * time.Second,
```
Comment (michaljurecko) [inline internal/pkg/utils/etcdhelper/fortest.go:120]: Fix for etcd in CI, test helper.

File: internal/pkg/service/stream/storage/level/local/writer/csv/csv_test.go [hunk 5]
```diff
@@ -308,7 +308,7 @@ func newTestCase(comp fileCompression, syncMode disksync.Mode, syncWait bool, pa
 	}
 
 	return &testcase.WriterTestCase{
-		Name:     fmt.Sprintf("compression=%s,syncMode=%s,wait=%t,parallelWrite=%t", comp.Name, syncMode, syncWait, parallelWrite),
+		Name:     fmt.Sprintf("compression-%s-sync-%s-wait-%t-parallel-%t", comp.Name, syncMode, syncWait, parallelWrite),
 		FileType: model.FileTypeCSV,
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/writer/csv/csv_test.go:321]: Test name is used as a part of the temp dir name, so I removed Windows sensitive characters.

File: internal/pkg/service/stream/storage/level/local/reader/readchain/chain_test.go [hunk 6]
```diff
@@ -125,6 +125,9 @@ func TestChain_UnwrapFile_Ok(t *testing.T) {
 	file, ok = chain.UnwrapFile()
 	assert.Nil(t, file)
 	assert.False(t, ok)
+
+	// Close file
+	assert.NoError(t, expectedFile.Close())
 }
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/reader/readchain/chain_test.go:131]: In the Linux, there is no problem if a file is not closed in a test (file descriptors are automatically freed after the program ends).

But in the Windows, it fails, the `t.TempDir -> t.Cleanup` helper cannot delete the temp dir after the test.

So, I have to close all resources also in tests.

File: internal/pkg/service/stream/storage/level/local/writer/volume/writer_test.go [hunk 7]
```diff
@@ -212,7 +226,6 @@ ghi,jkl,789
 {"level":"debug","message":"file synced"}
 {"level":"debug","message":"chain closed"}
 {"level":"debug","message":"closed file"}
-{"level":"info","message":"closed volume"}
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/writer/volume/writer_test.go:215]: Volume is closed after the test by the `t.Cleanup`, so it is not part of the checked logs.

**Review by hosekpeter — APPROVED** [review]
